packages/server/src/queue/queue.controller.ts

```diff
@@ -49,19 +49,11 @@ export class QueueController {
   @Roles(Role.TA, Role.PROFESSOR, Role.STUDENT)
   async getQuestions(
     @Param('queueId') queueId: number,
-    @QueueRole() role: string,
-    @User() user: UserModel,
+    @QueueRole() role: Role,
+    @UserId() userId: number,
   ): Promise<ListQuestionsResponse> {
     const questions = await this.queueService.getQuestions(queueId);
-    if (role === Role.STUDENT) {
-      return questions.map((question) => {
-        if (question.creator.id !== user.id) {
-          question.creator = pick(question.creator, ['id']);
-        }
-        return question;
-      });
-    }
-    return questions;
+    return this.queueService.anonymizeQuestions(questions, userId, role);
```

Comment:
I like that you abstracted this but it might be more readable if you left the `if (role === Role.STUDENT) {` check in there so that the `this.queueService.anonymizeQuestions` would only be anonymizing the data it if the student was the one hitting the endpoint. 

It's a minor thing tho, so if you want to leave it that's totally fine.